.github/workflows/cypress.yml

```diff
@@ -0,0 +1,156 @@
+name: Cypress Tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - alpha
+      - beta
+      - stable
+
+env:
+  CYPRESS_TEST_USER_EMAIL: "dev@mr-beam.org"
+  CYPRESS_TEST_USER_PASSWORD: "a"
+  MRBEAM_PLUGIN_URL: "http://plugin-address.org:5002/"
+  MRBEAM_PLUGIN_DOMAIN_NAME: "plugin-address.org"
+
+jobs:
+  mrb-docker:
+    runs-on: ubuntu-latest
+    outputs:
+      domain: ${{ env.MRBEAM_PLUGIN_DOMAIN_NAME }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build and export
+        uses: docker/build-push-action@v3
+        with:
+          push: false
+          context: .
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          file: ./Dockerfile
+          tags: mrbeam_plugin:latest
+          outputs: type=docker,dest=/tmp/image.tar
+      - name: Upload MrBeamPlugin image
+        uses: actions/upload-artifact@v3
+        with:
+          name: mrbeam_plugin
+          path: /tmp/image.tar
+
+  ui-chrome-tests:
+    runs-on: ubuntu-latest
+    permissions: write-all
+    container:
+      image: cypress/browsers:node16.17.0-chrome106
+      options: --add-host=${{ needs.mrb-docker.outputs.domain }}:host-gateway
+      volumes:
+        - /var/run/docker.sock:/var/run/docker.sock
```

Comment:
we could add a comment here on why has this been added

Reply:
Done